test/test_enum.py

```diff
@@ -116,3 +116,47 @@ def test_non_str_keys_enum(self):
         self.assertEqual(
             orjson.dumps({IntEnum.ONE: 1}, option=orjson.OPT_NON_STR_KEYS), b'{"1":1}'
         )
+
+class EnumPassthroughTests(unittest.TestCase):
+    def test_passthrough_flag_enum(self):
+        with self.assertRaises(orjson.JSONEncodeError):
+            orjson.dumps(FlagEnum.ONE, option=orjson.OPT_PASSTHROUGH_ENUM)
+
+    def test_passthrough_float_enum(self):
+        with self.assertRaises(orjson.JSONEncodeError):
+            orjson.dumps(FloatEnum.ONE, option=orjson.OPT_PASSTHROUGH_ENUM)
+
+    def test_passthrough_int_enum(self):
+        """
+        IntEnum is a subclass of int so should be unaffected by the passthrough flag
+        """
+        self.assertEqual(orjson.dumps(IntEnum.ONE, option=orjson.OPT_PASSTHROUGH_ENUM), b"1")
+
+    def test_passthrough_str_enum(self):
+        """
+        StrEnum is a subclass of str so should be unaffected by the passthrough flag
+        """
+        self.assertEqual(orjson.dumps(StrEnum.AAA, option=orjson.OPT_PASSTHROUGH_ENUM), b'"aaa"')
+
+    def test_passthrough_auto_enum(self):
+        self.assertEqual(orjson.dumps(AutoEnum.A, option=orjson.OPT_PASSTHROUGH_ENUM), b'"a"')
+
+    def test_passthrough_int_enum_enum(self):
+        """
+        IntEnumEnum is a subclass of int so should be unaffected by the passthrough flag
+        """
+        self.assertEqual(orjson.dumps(IntEnumEnum.ONE, option=orjson.OPT_PASSTHROUGH_ENUM), b"1")
+
+    def test_passthrough_int_flag_enum(self):
+        """
+        IntFlagEnum is a subclass of int so should be unaffected by the passthrough flag
+        """
+        self.assertEqual(orjson.dumps(IntFlagEnum.ONE, option=orjson.OPT_PASSTHROUGH_ENUM), b"1")
+
+    def test_passthrough_default_enum_default(self):
+        def default(obj):
+            return str(obj)
+
+        names = enum.Enum("names", ("john", "alice"))
+        self.assertEqual(orjson.dumps(names.alice, option=orjson.OPT_PASSTHROUGH_ENUM, default=default), b'"names.alice"')
```

Comment:
Should "names.alice" be the default behavior? I'm not familiar.

Reply:
I do not think so. I think how its implemented is fine. 

If you look at https://docs.python.org/3/library/json.html#json.JSONEncoder
You will see

```
Python | Json
...
int, float, int- & float-derived Enums | number
```

So the default serialization is correct.